### PR TITLE
feat: Add sparse float vector support to PyMilvus

### PR DESCRIPTION
**Pull request by zhengbuqian**
This PR adds the basic sparse float vector support to pymilvus. 

The following types of input are supported:

- any of the `scipy.sparse` matrix/array class family
- list of dicts: `[{0: 0.3, 20: 0.5}, {4: 0.4}, ...]`
- list of lists of tuples: `[[(0, 0.3), (20, 0.5)], [(4, 0.4)], ...]`

Sparse float vector support tracking issue: https://github.com/milvus-io/milvus/issues/29419

### CODE REVIEW

**Comment by zhengbuqian** [comment]
/hold

reviewer please review, but I'll wait until all milvus PRs(expecting by Wednesday) before merging this one.

File: pymilvus/client/prepare.py [hunk 1]
```diff
@@ -456,11 +456,11 @@ def _pre_batch_check(
         fields_info: Any,
     ):
         for entity in entities:
-            if (
-                not entity.get("name", None)
-                or not entity.get("values", None)
```
Comment (czs007) [inline pymilvus/client/prepare.py:461]: or entity.get("values", None) is None

**Comment by XuanYang-cn** [comment]
hold until milvus PR is merged

**Comment by zhengbuqian** [comment]
/unhold

all required milvus PRs are merged

**Pull request merged**